packages/java/endpoint/src/test/java/dev/hilla/auth/EndpointAccessCheckerTest.java

```diff
@@ -48,14 +48,24 @@ private void createAnonymousContext() {
         when(requestMock.getUserPrincipal()).thenReturn(null);
     }
 
-    private void shouldPass(Class<?> test) throws Exception {
-        Method method = test.getMethod("test");
-        assertNull(checker.check(method, requestMock));
+    private void assertAccessGranted(Class<?> test, String methodName)
+            throws Exception {
+        Method method = test.getMethod(methodName);
+        if (method.getDeclaringClass().equals(test)) {
```

Comment:
Why is this logic needed here? It is like you code the test according to how the code should work?

Reply:
This part is not related to the `EndpointAccessChecker` logic. This logic is tested in`EndpointInvokerTest`.